packages/framework/aqueduct/src/data-objects/treeDataObject.ts

```diff
@@ -0,0 +1,131 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import type { ISharedObject } from "@fluidframework/shared-object-base/internal";
+import {
+	type ImplicitFieldSchema,
+	type InsertableTreeFieldFromImplicitField,
+	type ITree,
+	SharedTree,
+	type TreeView,
+	type TreeViewConfiguration,
+} from "@fluidframework/tree/internal";
+
+import { PureDataObject } from "./pureDataObject.js";
+
+/**
+ * Channel ID of {@link TreeDataObject}'s root {@link @fluidframework/tree#SharedTree}.
+ */
+const treeChannelId = "root";
+
+/**
+ * {@link @fluidframework/tree#SharedTree}-backed {@link PureDataObject | data object}.
+ *
+ * @internal
```

Comment:
I don't like exposing (from the package) more classes intended for subclassing by users. I think moving the compositional pattern used in tree-react-utils here, where you provide the config and get back the SharedObjectKind would be a better APi than an abstract class which is exported

Reply:
I agree. But one of the goals here is to make it easy to substitute these new types into existing patterns on their side, so I've generally erred on the side of parity with the existing types here.

That said, before we promote these APIs beyond `@internal`, I plan to have prototyped an end-to-end solution in Loop that enables their use. That would be a good time to evaluate how reasonable it would be to refactor this pattern.

I can at least leave a comment to this effect here.